spec/src/main/asciidoc/architecture.adoc

```diff
@@ -159,10 +183,24 @@ There is one `MetricRegistry` instance for each of the scopes listed in <<scopes
 Metrics can be added to or retrieved from the registry either using the `@Metric` annotation
 (see <<api-annotations, Metrics Annotations>>) or using the `MetricRegistry` object directly.
 
+A metric is uniquely identified by the `MetricRegistry` if the `MetricID` associated with the metric is unique. That is to say, there are no other metrics with the same combination of metric name and tags. However, all metrics of the same name must be of the same type otherwise an `IllegalArgumentException` will be thrown.
```

Comment:
should say when this IllegalArgumentException will be thrown... I presume that's when code attempts to register the new metric that is inconsistent with they type of the existing metric